### DISCUSSION TITLE
Has anyone used a different Modbus converter?

### DISCUSSION THREAD

**Discussion by ndrinta** [StealthChesnut/HA-FoxESS-Modbus#88] [Q&A]
Hi folks, 

I was wondering if anyone here uses (or is thinking about using) a different Ethernet to RS485 (Modbus compatible) device than the W610.
I was thinking about the [USR-TCP232-410S](https://www.pusr.com/products/modbus-serial-to-ethernet-converters-usr-tcp232-410s.html) or [RS485_TO_ETH_(B)](https://www.waveshare.com/wiki/RS485_TO_ETH_(B)), being a little cheaper and, at least on paper, offering exactly the same features (except WIFi, which some of us wouldn't use anyways).

Let me know! 😃

**Comment by spmaid** [comment]
I am still able to use the LAN port, I guess due to not getting a firmware update that blocks it, I have however considered getting the serial adapter, just incase they do block, so I can quickly convert it over, but, I'd be more interested in an ethernet adapter over wireless for sure.  Wireless shouldn't generally be an issue for me, but I always prefer to cable where I can.

**Reply by TonyM1958** [reply]
I got this to work reliably after some fiddling around. Fewer sensors than RS485 but easier to connect. I found the ethernet port locked up randomly even with Manager firmware 1.57, but adding 'delay:1' to the modbus TCP settings seems to have fixed this so it's 100% reliable.

**Comment by phips** [comment]
I used [one of these](https://www.amazon.co.uk/dp/B078X5H8H7), connected to a USB port of my Pi (which is running HomeAssistant). It was simple, worked straight away, and cost 13 quid. Bargain!

**Comment by calum-mcfarlane** [comment]
@Ndrinta I've just got a 410S so I'll let you know how I get on once I've got it working. I'm connected via the LAN port at the moment but I want to be able to read / write the charging window registers and also not be worried if the next firmware update  will bork my connection to HA...

**Comment by calum-mcfarlane** [comment]
@Ndrinta I got the 410S working this evening, did a bit of a writeup here https://foxesscommunity.com/viewtopic.php?p=809#p809

**Reply by calum-mcfarlane** [reply]
As a bit more of a report on this, the 410S seems to have pretty good readout speeds compared to the W610 (see herehttps://github.com/nathanmarlor/foxess_modbus/discussions/40). The best (and cheapest!) option seems to be the Waveshare unit [here](https://thepihut.com/products/industrial-serial-server-rs485-to-rj45-ethernet) but I found out about that after I'd ordered my 410S.

**Reply by spmaid** [reply]
Good read! I managed to get a PoE Waveshare device via Amazon, just incase my LAN port stopped working due to a firmware update but so for Fox support haven’t touched that. I’m supposedly on the latest firmware for the batteries too but still get the massive jumps on SoC 🙄 I’ve been told one battery is faulty and will be replaced under RMA, I strongly suspect one other battery is not right either.

**Comment by TonyM1958** [comment]
I got down to £5.99 for USB to RS485 converter that just works, thanks to a hint from Brian Ely. https://www.amazon.co.uk/RUNCCI-RS485-Converter-Adapter-Window/dp/B07K3V381Z. My HA host is next to inverter, so this just plugs into a USB port and 1m of UTP cable to connect it. Obviously, not ethernet, but much cheaper...

**Reply by calum-mcfarlane** [reply]
Yep if you can connect directly then you have other (much cheaper!) options than the Modbus TCP bridges. Sadly my HA host is two stories up from the inverter so networking the setup was the only choice. The RS485 connection (whether via TCP bridge or serial to USB adapter) is the better way to go because you have the possibility (assuming suitable software) of writing to the registers and controlling the inverter that way - as far as I know this isn't possible with the ethernet connection.

**Comment by ndrinta** [comment]
I tried with the Waveshare RS485 TO ETH (B) and works like a charm, you can find a guide on how to set it up [here](https://github.com/StealthChesnut/HA-FoxESS-Modbus/discussions/122).
